swagger/specification/applications/resource-manager/Applications.Core/preview/2023-10-01-preview/openapi.json

```diff
@@ -2795,7 +2795,12 @@
     "BicepRecipeProperties": {
       "type": "object",
       "description": "Represents Bicep recipe properties.",
-      "properties": {},
+      "properties": {
+        "insecureHttp": {
+          "type": "boolean",
+          "description": "Allows insecure connections to registry without SSL check"
```

Comment:
```suggestion
          "description": "Allows an insecure connection to a Bicep registry without doing an SSL check. Used commonly for locally-hosted registries. Defaults to false (require SSL)."
```

Reply:
I personally find the term "SSL check" very misleading. There's an incredible amount of nuance here and simplifying the description to "check SSL" vs "don't check SSL" isn't accurate.

If we limit ourselves to common scenarios for the web, there's actually a third option:

- Use HTTP (option we're adding)
- Use HTTPS and verify the certificate's authenticity (Default)
- Use HTTPS and accept an invalid or self-signed certificate (option we're not adding)

We're not adding support for the third option here, but language like "SSL Check" strongly suggests we are.

I think `Allow` is also misleading. We're not trying HTTPS and then falling back to HTTP. We're choosing up front to do one or the other.

Since this is a security feature I'd suggest being really clear. We don't need to optimize for simplicity here, we're optimizing for clarity. Something like:

> Connect to the Bicep registry using HTTP (not-HTTPS). This should be used when the registry is known not to support HTTPS, for example in a locally-hosted registry. Defaults to false (use HTTPS/TLS).

We should avoid using the term SSL in favor of TLS unless we need to be consistent with another system that uses the term SSL. SSL has been superseded by TLS (it's a bigger, more inclusive term). There are no new releases of SSL, but TLS is actively being developed.